### PR TITLE
[Docs] Add migration note about expanded fields limit

### PR DESCRIPTION
**Pull request by cbuescher**
Adds a note to warn users about the limit introduced in #26541.

### CODE REVIEW

File: docs/reference/migration/migrate_7_0/search.asciidoc [hunk 1]
```diff
@@ -106,6 +106,14 @@ To safeguard against this, the maximum length of regex that can be used in a
 Regexp Query request has been limited to 1000. This default maximum can be changed
 for a particular index with the index setting `index.max_regex_length`.
 
+[float]
+==== Limiting the number of auto-expanded fields
+
+Executing queries that use automatic expansion of fields (e.g. `query_string`, `simple_query_string`
+or `multi_match`) can have performance issues for indices with a large numbers of fields.
+To safeguard against this, a fixed limit of a maximum of 1024 fields has been introduced
+for queries using the "all fields" mode ("default_field": "*").
```
Comment (jimczi) [inline docs/reference/migration/migrate_7_0/search.asciidoc]: It is applied to any expansion so not only `*` but also `a*` or `a,b,c,d*`

File: docs/reference/migration/migrate_7_0/search.asciidoc [hunk 2]
```diff
@@ -106,6 +106,14 @@ To safeguard against this, the maximum length of regex that can be used in a
 Regexp Query request has been limited to 1000. This default maximum can be changed
 for a particular index with the index setting `index.max_regex_length`.
 
+[float]
+==== Limiting the number of auto-expanded fields
+
+Executing queries that use automatic expansion of fields (e.g. `query_string`, `simple_query_string`
+or `multi_match`) can have performance issues for indices with a large numbers of fields.
+To safeguard against this, a fixed limit of a maximum of 1024 fields has been introduced
```
Comment (jimczi) [inline docs/reference/migration/migrate_7_0/search.asciidoc]: `an hard limit of 1024 fields` ?

**Review by jimczi — CHANGES_REQUESTED** [review]
Thanks @cbuescher . I left some comments

**Comment by cbuescher** [comment]
@jimczi I updated the text according to your comments. Would you be able to take another look?

**Review by jimczi — APPROVED** [review]
Thanks for updating @cbuescher !

**Pull request merged**